### PR TITLE
Preparing the alpha release

### PR DESCRIPTION
**Pull request by eistrati**

### CODE REVIEW

File: atm.module [hunk 1]
```diff
@@ -83,16 +74,144 @@ function atm_theme($existing, $type, $theme, $path) {
...
+ */
 function atm_admin_config_page_preprocess(&$variables) {
   $variables['module_dir'] = drupal_get_path('module', 'atm');
 }
 
 /**
+ * Implements hook_cron().
+ */
+function atm_cron() {
+  if (REQUEST_TIME - \Drupal::state()->get('google_atm.last_cache') >= 86400) {
```
Comment (AlexanderC) [inline atm.module:167]: Is there a reason the state is prefixed `google_atm` and not just `atm`?

Comment (AlexanderC) [inline atm.module:167]: Please move `86400` ttl value to config

Comment (AlexanderC) [reply atm.module:167]: What's the difference between 'google_atm.last_cache' and `atm.last_cache`?

File: config/install/atm.settings.yml [hunk 2]
```diff
@@ -1,8 +1,13 @@
 register_url: https://www.adtechmedia.io/admin/accounts/signup
```
Comment (AlexanderC) [inline config/install/atm.settings.yml]: Please create an `atm.settings.dev.yml` and load it with all development endpoints when localhost used

File: src/Controller/AtmApiController.php [hunk 3]
```diff
@@ -2,24 +2,135 @@
...
+   */
+  public function getSwJs() {
+    $response = new Response('', 200, ['Content-Type' => 'application/javascript']);
+
+    try {
+      $httpResponse = $this->httpClient->get($this->atmApiHelper->get('sw_js_file'));
+      $response->setContent($httpResponse->getBody()->getContents());
+    }
+    catch (ClientException $e) {
+
```
Comment (AlexanderC) [inline src/Controller/AtmApiController.php:94]: Please handle the exception properly

File: src/Helper/AtmApiHelper.php [hunk 4]
```diff
@@ -178,76 +244,136 @@ public function getRevenueModelList() {
...
+          "Directory `atm` wasn't created. Check permissions on public files directory. This directory must exist and be writable by Drupal"
+        );
+      }
     }
 
-    $script = file_get_contents($remote);
+    $realpath = $this->fileSystem->realpath($path_schema);
+
+    $script = file_get_contents($remote . '?' . microtime());
     $script = gzdecode($script);
```
Comment (AlexanderC) [inline src/Helper/AtmApiHelper.php:267]: What happens if the script is not gziped?

File: src/Helper/AtmApiHelper.php [hunk 5]
```diff
@@ -178,76 +244,136 @@ public function getRevenueModelList() {
...
-    .atm-base-modal .atm-footer {
-      background-color: $fbg;
-      border: $fb;
-    }
-    
+  /**
+   * Get CSS template for atm-modal.
+   */
+  public function getCssTemplate() {
+    return "
```
Comment (AlexanderC) [inline src/Helper/AtmApiHelper.php:338]: Could we move css to static assets?

**Review by AlexanderC — CHANGES_REQUESTED** [review]

File: atm.module (hunk 1 above)
Comment (azzaromax) [reply atm.module:167]: Need rename `google_atm.last_cache` to `atm.last_cache`
My error

File: config/install/atm.settings.yml (hunk 2 above)
Comment (azzaromax) [reply config/install/atm.settings.yml]: It's impossible
Drupal load config/install/atm.settings.yml by pattern "{module_name}.settings.yml"

File: src/Controller/AtmApiController.php (hunk 3 above)
Comment (azzaromax) [reply src/Controller/AtmApiController.php:94]: Ok, I can write exception to log

File: src/Helper/AtmApiHelper.php (hunk 5 above)
Comment (azzaromax) [reply src/Helper/AtmApiHelper.php:338]: And what's the point? This css contains a template and not a full css

**Pull request merged**